strawberry/ext/mypy_plugin.py

```diff
@@ -245,17 +253,114 @@ def enum_hook(ctx: DynamicClassDefContext) -> None:
+    """Adds a new method to a class definition.
+    Edited to add is_static for
+    https://github.com/python/mypy/blob/master/mypy/plugins/common.py
+    because couldn't figure out how to add static methods otherwise"""
+    info = cls.info
+
+    # First remove any previously generated methods with the same name
+    # to avoid clashes and problems in the semantic analyzer.
+    if name in info.names:
+        sym = info.names[name]
+        if sym.plugin_generated and isinstance(sym.node, FuncDef):
+            cls.defs.body.remove(sym.node)
+
+    self_type = self_type or fill_typevars(info)
+    function_type = api.named_type("__builtins__.function")
+
+    if not is_static:
+        args = [Argument(Var("self"), self_type, None, ARG_POS)] + args
+    arg_types, arg_names, arg_kinds = [], [], []
+    for arg in args:
+        assert arg.type_annotation, "All arguments must be fully typed."
+        arg_types.append(arg.type_annotation)
+        arg_names.append(arg.variable.name)
+        arg_kinds.append(arg.kind)
+
+    signature = CallableType(
+        arg_types, arg_kinds, arg_names, return_type, function_type
+    )
+    if tvar_def:
+        signature.variables = [tvar_def]
+
+    func = FuncDef(name, args, Block([PassStmt()]))
+    if is_static:
+        func.is_static = True
+    func.info = info
+    func.type = set_callable_name(signature, func)
+    func._fullname = info.fullname + "." + name
+    func.line = info.line
+
+    # NOTE: we would like the plugin generated node to dominate, but we still
+    # need to keep any existing definitions so they get semantically analyzed.
+    if name in info.names:
+        # Get a nice unique name instead.
+        r_name = get_unique_redefinition_name(name, info.names)
+        info.names[r_name] = info.names[name]
+
+    info.names[name] = SymbolTableNode(MDEF, func, plugin_generated=True)
+    info.defn.defs.body.append(func)
+
+
+def strawberry_pydantic_class_callback(ctx: ClassDefContext) -> None:
     # in future we want to have a proper pydantic plugin, but for now
-    # let's fallback to any, some resources are here:
+    # let's fallback to **kwargs for __init__, some resources are here:
     # https://github.com/samuelcolvin/pydantic/blob/master/pydantic/mypy.py
     # >>> model_index = ctx.cls.decorators[0].arg_names.index("model")
     # >>> model_name = ctx.cls.decorators[0].args[model_index].name
 
     # >>> model_type = ctx.api.named_type("UserModel")
     # >>> model_type = ctx.api.lookup(model_name, Context())
 
-    ctx.cls.info.fallback_to_any = True
+    model_expression = _get_argument(call=ctx.reason, name="model")  # type: ignore
+    if model_expression is None:
+        ctx.api.fail("model argument in decorator failed to be parsed", ctx.reason)
+
+    else:
+        # Add __init__
+        init_args = [
+            Argument(Var("kwargs"), AnyType(TypeOfAny.explicit), None, ARG_STAR2)
+        ]
+        add_method(ctx, "__init__", init_args, NoneType())
```

Comment:
we still don't have any mypy validation for the arguments. we'll just accept kwargs for now.